DS-Session-2-Assignment/src/question2/QueueImplementation.java

```diff
@@ -0,0 +1,118 @@
+package question2;
+
+/**
+ * This class implements all the methods of the Queue interface
+ * @author Jugal
+ * Dated 07/31/2019
+ */
+public class QueueImplementation implements Queue{
+	
+	int queueSize = 5;
+	int[] queue = new int[queueSize];
+	int start = -1;
+	int end = -1;
+	
+	/**
+	 * This method will add data in the queue in the end
+	 *  @param data is the data to be inserted
+	 */
+	@Override
+	public void enqueue(int data) throws Exception {
+		if(isQueueEmpty()) //If queue is empty set start and end to 0
+		{
+			start = 0;
+			end = 0;
+		}
+		else if(isQueueFull())
+		{
+			throw new Exception("Queue is full");
```

Comment:
do not throw Exception, just print the message to the user and continue your program.